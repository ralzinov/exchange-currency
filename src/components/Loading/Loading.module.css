.host {
    justify-content: center;
    display: flex;
    align-items: center;
    flex: 1;
}

@keyframes spin {
    0% {
        -webkit-transform: rotate(0);
        transform: rotate(0);
    }
    100% {
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg);
    }
}

.spinner {
    position: relative;
}

.spinner div {
    box-sizing: border-box;
}

.spinner > div {
    position: absolute;
    width: 168px;
    height: 168px;
    top: 16px;
    left: 16px;
    border-radius: 50%;
    border-style: solid;
    border-width: 8px;
    border-color: #28292f transparent #28292f transparent;
    -webkit-animation: spin 1.6s linear infinite;
    animation: spin 1.6s linear infinite;
}

.spinner > div:nth-child(2) {
    border-color: transparent;
}

.spinner > div:nth-child(2) div {
    position: absolute;
    width: 100%;
    height: 100%;
    -webkit-transform: rotate(45deg);
    transform: rotate(45deg);
}

.spinner > div:nth-child(2) div:before,
.spinner > div:nth-child(2) div:after {
    content: "";
    display: block;
    position: absolute;
    width: 8px;
    height: 8px;
    top: -8px;
    left: 72px;
    background: #28292f;
    border-radius: 50%;
    box-shadow: 0 160px 0 0 #28292f;
}

.spinner > div:nth-child(2) div:after {
    left: -8px;
    top: 72px;
    box-shadow: 160px 0 0 0 #28292f;
}

.spinner {
    width: 64px;
    height: 64px;
    -webkit-transform: translate(-32px, -32px) scale(0.32) translate(32px, 32px);
    transform: translate(-32px, -32px) scale(0.32) translate(32px, 32px);
}
