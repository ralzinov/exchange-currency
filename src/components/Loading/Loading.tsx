import * as React from 'react';
import styles from './Loading.module.css';

export const Loading = () => {
    return (
        <div className={styles.host}>
            <div className={styles.spinner}>
                <div/><div><div/></div>
            </div>
        </div>
    );
};
