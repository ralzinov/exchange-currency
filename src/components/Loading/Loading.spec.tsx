import React from 'react';
import ReactDOM from 'react-dom';
import {Loading} from './Loading';
import {create} from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Loading/>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should render with children', () => {
    const button = create(
        <Loading/>
    );
    expect(button.toJSON()).toMatchSnapshot();
});
