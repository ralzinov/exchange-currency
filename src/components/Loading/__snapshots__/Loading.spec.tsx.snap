// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with children 1`] = `
<div
  className="host"
>
  <div
    className="spinner"
  >
    <div />
    <div>
      <div />
    </div>
  </div>
</div>
`;
